### PR TITLE
Add ARM64 support and cache Docker images

### PR DESCRIPTION
**Pull request by connorv001**
- Add support for ARM64 platform in build-and-publish-docker.yml
- Allow caching of Docker images in build-and-publish-docker.yml

[.github/workflows/build-and-publish-docker.yml]
- Add support for ARM64 platform
- Allow caching of Docker images

### CODE REVIEW

**Comment by Kav-K** [comment]
@cooperlees

**Comment by Hikari-Haru** [comment]
Hey, it doesn't seem like rust is properly exported to the path, it fails on arm64 build due to missing rust

**Comment by Hikari-Haru** [comment]
I also don't understand why you have set the environment variables in the docker-compose config instead of using a volume with the env file with `env_file:/opt/gpt3discord/etc/environment` as it'll load it from there too in code. You also haven't set DATA_DIR and SHARE_DIR as environment variables so that the volumes added actually do anything in compose to begin with. Have you tested that persistence works with what you've added?

**Comment by Hikari-Haru** [comment]
Excuse me, what?

File: .github/workflows/build-and-publish-docker.yml [hunk 1]
```diff
@@ -1,96 +1,97 @@
...
+name: Docker
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+on:
+  schedule:
+    - cron: '28 2 * * *'
```
Comment (cooperlees) [inline .github/workflows/build-and-publish-docker.yml:10]: Why do we cron it? Seems a waste of resource?

File: .github/workflows/build-and-publish-docker.yml [hunk 2]
```diff
@@ -1,96 +1,97 @@
...
+
+      # Workaround: https://github.com/docker/build-push-action/issues/461
+      - name: Setup Docker buildx
+        uses: docker/setup-buildx-action@79abd3f86f79a9d68a23c75a09a9a85889262adf
+
+      # Login against a Docker registry except on PR
+      # https://github.com/docker/login-action
+      - name: Log into registry ${{ env.REGISTRY }}
+        if: github.event_name != 'pull_request'
+        uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
```
Comment (cooperlees) [inline .github/workflows/build-and-publish-docker.yml:57]: Do we need this specific commit? Can we use a specific version and have CI cut new versions when they come out?


Same with others using different commits ...

**Review by cooperlees — COMMENTED** [review]
This does a lot more then just add ARM64 and cache Docker images ... Could It possible be divided up into smaller more reviewable PRs ?

**Comment by Hikari-Haru** [comment]
> This does a lot more then just add ARM64 and cache Docker images ... Could It possible be divided up into smaller more reviewable PRs ?

All of the last commit is no changes, only marked changes, probably git CRLF stuff

File: .github/workflows/build-and-publish-docker.yml (hunk 1 above)
Comment (Kav-K) [reply .github/workflows/build-and-publish-docker.yml:10]: agreed, let's get rid of the cron

**Comment by Kav-K** [comment]
What happened with the file history with this PR?

**Pull request closed without merging**